website/source/docs/job-specification/template.html.md

```diff
@@ -0,0 +1,108 @@
+---
+layout: "docs"
+page_title: "template Stanza - Job Specification"
+sidebar_current: "docs-job-specification-template"
+description: |-
+  The "template" block instantiates an instance of a template renderer. This
+  creates a convenient way to ship configuration files that are populated from
+  Consul data, Vault secrets, or just general configurations within a Nomad
+  task.
+---
+
+# `template` Stanza
+
+<table class="table table-bordered table-striped">
+  <tr>
+    <th width="120">Placement</th>
+    <td>
+      <code>job -> group -> task -> **template**</code>
+    </td>
+  </tr>
+</table>
+
+The `template` block instantiates an instance of a template renderer. This
+creates a convenient way to ship configuration files that are populated from
+Consul data, Vault secrets, or just general configurations within a Nomad task.
+
+```hcl
+job "docs" {
+  group "example" {
+    task "server" {
+      template {
+        source        = "local/redis.conf.tpl"
+        destination   = "local/redis.conf"
+        change_mode   = "signal"
+        change_signal = "SIGNINT"
+      }
+    }
+  }
+}
+```
+
+Nomad is utilizes a tool called [Consul Template][ct]. For a full list of the
+API template functions, please see the [Consul Template README][ct].
+
+## `template` Parameters
+
+- `source` `(string: "")` - Specifies the path to the template to be rendered.
+  One of `source` or `data` must be specified, but not both. This source can
+  optionally be fetched using an [`artifact`][artifact] resource.
+
+- `destination` `(string: required)` - Specifies the location where the resulting template should be rendered, relative to the task directory.
+
+- `data` `(string: "")` - Specifies the raw template to execute. One of `source`
+  or `data` must be specified, but not both. This is useful for smaller
+  templates, but we recommend using `source` for larger templates.
+
+- `change_mode` `(string: "restart")` - Specifies the behavior Nomad should take if the Vault token changes. The possible values are:
+
+  - `"noop"` - take no action (continue running the task)
+  - `"restart"` - restart the task as per the task's [restart policy][restart]
+  - `"signal"` - send a configurable signal to the task
+
+- `change_signal` `(string: "")` - Specifies the signal to send to the task as a
+  string like `"SIGUSR1"` or `"SIGINT"`. This option is required if the
+  `change_mode` is `signal`.
+
+- `splay` `(string: "5s")` - Specifies a random amount of time to wait between
```

Comment:
quiescence made a huge difference for us (bownty) when running CT, since our clusters are periodically very busy (5-10s) and then idle for hours and hours :)